### ISSUE TITLE
Crash in protocol manager when starting a race

### ISSUE THREAD

**Issue by deveee** [supertuxkart/stk-code#2658]
When I start a race, I get following crash:
```
(gdb) bt
#0  0x000000010057c9d2 in MainLoop::run (this=0x6010acd30)
    at /cygdrive/d/deve/stk-code/src/main_loop.cpp:298
#1  0x000000010057a694 in main (argc=1, argv=0xffffcc10)
    at /cygdrive/d/deve/stk-code/src/main.cpp:1722
(gdb) bt full
#0  0x000000010057c9d2 in MainLoop::run (this=0x6010acd30)
    at /cygdrive/d/deve/stk-code/src/main_loop.cpp:298
        protocol = 0x605c3d9b0
        dt = 0.0410000011
        device = 0x6001af510
#1  0x000000010057a694 in main (argc=1, argv=0xffffcc10)
    at /cygdrive/d/deve/stk-code/src/main.cpp:1722
        s = {static npos = <optimized out>,
          _M_dataplus = {<std::allocator<char>> = {<__gnu_cxx::new_allocator<char>> = {<No data fields>}, <No data fields>},
            _M_p = 0x3dd12bbf8 <cygstdc++-6!_ZNSs4_Rep20_S_empty_rep_storageE+24> ""}}
        materials_file = {static npos = <optimized out>,
          _M_dataplus = {<std::allocator<char>> = {<__gnu_cxx::new_allocator<char>> = {<No data fields>}, <No data fields>},
            _M_p = 0x6065f03f8 "../data/../../stk-assets/models/materials.xml"}}
        important_message = {array = 0x606de6750 L"", allocated = 1, used = 1,
          allocator = {
            _vptr.irrAllocator = 0x100fc4800 <vtable for irr::core::irrAllocator<wchar_t>+16>}}
```

Seems to be related to this commit:
https://github.com/supertuxkart/stk-code/commit/b1f6632e5b23a5f20580d564b3e0712e2e354d9c

**Comment by hiker** [comment]
Yes, that 'fix' for a crash was indeed a bad idea (and I have already worked on a better version of this). If I can't fix this properly tonight, I'll revert this in a few hours.

Thanks!
  Joerg

**Comment by hiker** [comment]
Fixed in 9f4c842b6c2c523fe7ba61424e7865217e59ac4a, sorry for this crash. I'll move to work on a branch.

**Comment by deveee** [comment]
Works fine now. Thanks!
